.github/workflows/build_packages.yml

```diff
@@ -242,19 +177,205 @@ jobs:
+
+      - name: Build cuda-quantum image
+        id: cudaq_image
+        run: |
+          image_name=${{ needs.metadata.outputs.image_name }}
+          image_tag=${{ needs.metadata.outputs.image_tag }}
+          tar_archive=/tmp/cuda-quantum.tar
+
+          if ${{ steps.restore_openmpi.outcome != 'skipped' }}; then
+            base_image=`docker load --input ${{ needs.ompi_image.outputs.tar_archive }} | grep -o 'Loaded image: \S*:\S*' | cut -d ' ' -f 3`
+          else
+            base_image=ubuntu:22.04
+          fi
+
+          if ${{ steps.restore_cudaqdev.outcome != 'skipped' }}; then
+            cudaqdev_image=`docker load --input ${{ needs.cudaqdev_image.outputs.tar_archive }} | grep -o 'Loaded image: \S*:\S*' | cut -d ' ' -f 3`
+          else
+            docker pull ${{ needs.cudaqdev_deployment.outputs.image_hash }}
+            cudaqdev_image=${{ needs.cudaqdev_deployment.outputs.image_hash }}
+          fi
+
+          docker build -t $image_name:$image_tag -f docker/release/cudaq.Dockerfile . \
+            --build-arg cudaqdev_image=$cudaqdev_image \
+            --build-arg base_image=$base_image \
+            --build-arg release_version=$image_tag
+
+          rm -rf "${{ needs.ompi_image.outputs.tar_archive }}"
+          rm -rf "${{ needs.cudaqdev_image.outputs.tar_archive }}"
+          docker image rm $base_image
+          docker image rm $cudaqdev_image
+          docker image prune --force
+          docker container prune --force
+          docker volume prune --force
+          docker builder prune --all --force
+          docker save $image_name:$image_tag > $tar_archive
+
+          echo "tar_cache=tar-cudaq-${image_tag}-${{ github.sha }}" >> $GITHUB_OUTPUT
+          echo "tar_archive=$tar_archive" >> $GITHUB_OUTPUT
+
+      - name: Cache cuda-quantum image
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ steps.cudaq_image.outputs.tar_archive }}
+          key: ${{ steps.cudaq_image.outputs.tar_cache }}
+
+      - name: Validate cuda-quantum image
+        run: |
+          docker run --rm -dit --name cuda-quantum ${{ needs.metadata.outputs.image_name }}:${{ needs.metadata.outputs.image_tag }}
+          docker cp scripts/validate_container.sh cuda-quantum:"/home/cudaq/validate_container.sh"
+          docker exec -e TERM=xterm cuda-quantum bash validate_container.sh > /tmp/validation.out
+          docker stop cuda-quantum
+
+      - name: Create job summary
+        if: always() && !cancelled()
+        run: |
+          if [ -f /tmp/validation.out ]; then
+            echo "## Validation" >> $GITHUB_STEP_SUMMARY
+            echo "The validation of the cuda-quantum image produced the following output:" >> $GITHUB_STEP_SUMMARY
+            echo '```text' >> $GITHUB_STEP_SUMMARY
+            cat /tmp/validation.out >> $GITHUB_STEP_SUMMARY
+            echo '```' >> $GITHUB_STEP_SUMMARY
+          fi
+
+# FIXME: ONLY IF WE ARE NOT DEPLOYING FROM ELSEWHERE
```

Comment:
What does deploying from elsewhere mean?

Reply:
Ah, let me remove that comment. This was written with Option 1 in the PR description in mind. However, I've decided to defer that to a separate PR, to keep PRs smaller. Also, I think Option 2 might be preferable.